### ISSUE TITLE
Support searching users own contributions

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#5882]
### What is the user problem or growth opportunity you want to see solved?

The app's search functionality does not really allow searching their own contributions. It is a good feature to have particularly if users want to quickly find their contributions but have too many of them to quickly scroll and reach the specific one.

### How do you know that this problem exists today? Why is this important?

A user reported this on Play store. This is important as it addresses a user concern and help provide a better UX for all users when implemented. It's quite surprising  we didn't have this feature this far.

### Who will benefit from it?

All users of the app who want to quickly locate an existing upload of theirs would benefit from this.

**Comment by nicolas-raoul** [comment]
UI: not sure whether this feature should go in Explore (because it has other types of searches) or Contributions (because that list has what the user is looking for):

![Screenshot_20241022-114022.png](https://github.com/user-attachments/assets/ef426c4e-b205-4a24-a5c2-2b98be42861b)

![Screenshot_20241022-114357.png](https://github.com/user-attachments/assets/5b3cdc4d-039c-46ac-bd92-aaf64321b043)

![Screenshot_20241022-114022.png](https://github.com/user-attachments/assets/7908f3ac-0107-4816-8f4b-72d75df837d5)

**Comment by rohit9625** [comment]
The search functionality on contributions screen would be nice. It should also have a filter option to sort the latest or the oldest pictures and some other filters as well if possible.

**Comment by misaochan** [comment]
I agree that having the feature in Contributions would be nice. Most people wouldn't think to go to Explore to search their own uploads.

**Comment by nicolas-raoul** [comment]
This is the Commons app's home activity, so  users will expect that the search buton allows them to search all of Commons too.

So I suggest showing our existing search activity, but initially focussed on a new "My media" tab.

There would be 4 tabs:
My media / All media / Categories / Items
instead of the current 3:

![Screenshot_20241023-103347.png](https://github.com/user-attachments/assets/bc3e253e-91e4-4813-9bfb-67d44f0f90dc)

In Explore, tapping the search icon would lead to the same search activity, but initially focussed on "All media".

**Comment by prototyperspective** [comment]
This would be useful but are you talking about only uploads or also other contributions/edits. For the latter there is this to search edit summaries: https://sigma.toolforge.org/summary.py I think the the Commons desktop site doesn't have functionality to search one's own uploads, e.g. see https://phabricator.wikimedia.org/T188125 (one can only search for original works one has uploaded but not all files one has uploaded)

**Comment by nicolas-raoul** [comment]
@prototyperspective Actually yesterday I wanted to find a picture of a Pixel device that I had uploaded in the past. But I could not find it even by using the website. Would you know how to do that?

**Comment by chrisdebian** [comment]
Is this still in-scope for Milestone 5.2.0? It seems like we're running short of ideas to fix this, and at the moment, it would seems we have issues that are a higher priority.

Thanks,

Chris.

**Comment by neeldoshii** [comment]
* Taking it up. Just want to ask what's the conclusion should we have the search in Contribution Activity or in Explore as per https://github.com/commons-app/apps-android-commons/issues/5882#issuecomment-2430619787

**Comment by nicolas-raoul** [comment]
@neeldoshii Sorry for the delay! I recommend following https://github.com/commons-app/apps-android-commons/issues/5882#issuecomment-2430619787

**Comment by prototyperspective** [comment]
Couldn't one simply use https://sigma.toolforge.org/summary.py and link to the page with commonswiki and the username already filled in? Maybe there is an API for it. (Could not find any documentation for this and it seems like the code is just in the source code but not in any repo with code issues.)

**Comment by prototyperspective** [comment]
> [@prototyperspective](https://github.com/prototyperspective) Actually yesterday I wanted to find a picture of a Pixel device that I had uploaded in the past. But I could not find it even by using the website. Would you know how to do that?

There's probably some ways to do that. The tool I linked is only for the edit summary so if other things are to be searched that indeed wouldn't be helpful. That's what the phab issue is about where one would simply enter e.g. uploader:username pixel into the commons search to find it or enter the search term on the user's contributions page. This is the way I'd use without such: going to the user contributions page, expanding "Search for contributions" and check "Only show edits that are page creations", then make it show 500 results, then ctrl+f for pixel since that is probably somewhere in the file title.
